src/FamilyHubs.ServiceDirectory.Web/Pages/ServiceFilter/Index.cshtml.cs

```diff
@@ -57,10 +49,22 @@ public ServiceFilterModel(IServiceDirectoryClient serviceDirectoryClient, IPostc
     {
         _serviceDirectoryClient = serviceDirectoryClient;
         _postcodeLookup = postcodeLookup;
-        Filters = DefaultFilters;
         Services = Enumerable.Empty<Service>();
         OnlyShowOneFamilyHubAndHighlightIt = false;
         Pagination = new DontShowPagination();
+        
+        var taxonomies = _serviceDirectoryClient.GetTaxonomies().GetAwaiter().GetResult();
```

Comment:
This only needs to be done once at start-up, rather than for every request (applies to creating the IFilter list in general).

We should be careful calling `GetAwaiter().GetResult()` in a synchronous constructor. I think, if a SynchronizationContext gets added, this can cause an async deadlock. If we refactor to only set up the filters once, then we should use await/async for safety.

Reply:
moved this code from constructor to Get method, post is redirecting to get so it is covered in both scenarios